### ISSUE TITLE
[BC Idea]: Correct commit and error handling in codeunit Gen. Jnl.-Post Line OnAfterGLFinishPosting event

### ISSUE THREAD

**Issue by ernestasjuska** [microsoft/BCApps#2192]
### BC Idea Link

https://experience.dynamics.com/ideas/idea/?ideaid=5c4328ef-218a-ef11-9442-6045bdbf4815

### Description

Hi,

The general journal lines are only deleted after posting the batch. But the events from the codeunit Gen. Jnl.-Post Line allow commit before the posting ends. This means after a commit and then an error in the event subscriber, both posted entry and journal line remain in database.

For example, we stumbled upon a rare case when codeunit "CustEntry-Apply Posted Entries" calls codeunit "Payment Tolerance Management" that does the commit and then codeunit "CustEntry-Apply Posted Entries" shown an error.

To reproduce the issue, create a new app with an event subscriber like this:
```al
codeunit 50100 "BREAK Event Subscribers"
{
    InherentPermissions = X;

    [EventSubscriber(ObjectType::Codeunit, Codeunit::"Gen. Jnl.-Post Line", OnAfterGLFinishPosting, '', false, false)]
    local procedure "Gen. Jnl.-Post Line_OnAfterGLFinishPosting"()
    begin
        Commit();
        Error('err');
    end;
}
```
[PostingProblemApp.zip](https://github.com/user-attachments/files/17363271/PostingProblemApp.zip)

Posting a payment to customer using payments journal will show error message 'err'. The customer ledger entry will be created even though there was an error.

Please add attribute `[CommitBehavior(CommitBehavior::Ignore)]` on codeunit Gen. Jnl.-Post Line OnAfterGLFinishPosting event publisher.

Thanks.

Example correction - to demonstrate that it works (it can handle "if codeunit.run then"):
```al
codeunit 50100 "BREAK Event Subscribers"
{
    InherentPermissions = X;

    [EventSubscriber(ObjectType::Codeunit, Codeunit::"Gen. Jnl.-Post Line", OnAfterGLFinishPosting, '', false, false)]
    local procedure "Gen. Jnl.-Post Line_OnAfterGLFinishPosting"()
    begin
        OnFixedEvent();
    end;

    [IntegrationEvent(false, false)]
    [CommitBehavior(CommitBehavior::Ignore)] // NOTE: Add this to OnAfterGLFinishPosting
    local procedure OnFixedEvent()
    begin
    end;

    [EventSubscriber(ObjectType::Codeunit, Codeunit::"BREAK Event Subscribers", OnFixedEvent, '', false, false)]
    local procedure HandleOnFixedEvent()
    begin
        Commit();
        Error('err');
    end;
}
```

In general, I don't expect commits when using Gen. Jnl.-Post Line. So the attribute could be lifted to `procedure Code()`.

### I will provide the implementation for this BC Idea

- [x] I will provide the implementation for this BC Idea

**Comment by JesperSchulz** [comment]
Sounds fair, but since the code you are referring to is in the BaseApp, you'd have to suggest and fix this code in the BaseApp contribution pilot: https://github.com/microsoft/businesscentralapps. For access to the pilot, please fill out the following form: https://forms.office.com/r/E9wyE5a58f.

Would you like me to move the issue to the contribution pilot for you?

**IMPORTANT:** Microsoft is not going to address the reported issue for you. You will have to get the issue approved, and then push a pull request with the suggested changes yourself. If you would like Microsoft to address the issue, please create a support request. Learn more about our support channels here: https://aka.ms/bcsupport.

**Comment by ernestasjuska** [comment]
@JesperSchulz Yes, I would like to make code change in BaseApp. I submitted the form.
 
Thank you.
